### ISSUE TITLE
CODEC driver

### ISSUE THREAD

**Issue by Wolfvak** [linux-3ds/linux#11]
Currently there's a single "codec" device connected via the SPI bus and it controls touchscreen and circlepad stuff. In reality this driver also controls other registers responsible for sound (https://problemkaputt.de/gbatek.htm#3dstscregistersummary) and works similarly to the MCU or any other mfd, so it needs to be split off. However, the codec driver is also the one responsible for managing the swkbd which is pretty essential.

We need something like wifi or infrared to have proper communications with the system while debugging it (there are no UART ports exposed) before we can even think about ripping out swkbd and splitting the codec driver.

Marking as blocked by #6

**Comment by profi200** [comment]
"Reference" implementation:
https://github.com/profi200/open_agb_firm/blob/kernel_experiments/source/arm11/hardware/codec.c

It's a dumpster fire but that's to be expected with reverse engineered code i didn't have time to clean up yet.

**Comment by Wolfvak** [comment]
Yup, its listed in the wiki as such https://github.com/linux-3ds/linux/wiki/Driver-and-platform-support#spi-subdevices
